src/openrct2/ride/RideConstruction.cpp

```diff
@@ -638,6 +639,11 @@ void RideConstructionSetDefaultNextPiece()
             _currentTrackHasLiftHill = tileElement->AsTrack()->HasChain()
                 && ((slope != TrackPitch::Down25 && slope != TrackPitch::Down60)
                     || GetGameState().Cheats.enableChainLiftOnAllTrack);
+
+            if (TrackTypeHasSpeedSetting(tileElement->AsTrack()->GetTrackType()))
```

Comment:
Create an intermediate variable for `tileElement->AsTrack()` on line 638/639 and change the code to use that (for legibility).

Reply:
This caused a test to fail. Might need to be `auto*` or `auto&`; I'm not sure,

Reply:
auto&